ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -139,21 +225,23 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
                         }
                     }
 
-                    log.info("Streaming ingestion failed, trying again", e);
+                    log.info(String.format("Streaming ingestion failed, trying again after sleep of %s +~ %s seconds", retry.getCurrentSleepSecs(),
+                            retry.getMaxJitterSecs()), e);
+
                     try {
-                        streamSourceInfo.getStream().reset();
+                        managedSourceInfo.getStream().reset();
                     } catch (IOException ioException) {
-                        throw new IngestionClientException("Ingestion failed transiently but the stream isn't resettable therefore ingestion wasn't retried", ioException);
+                        throw new IngestionClientException("Failed to reset stream", ioException);
                     }
                 }
-            }
+            } while (retry.doBackoff());
```

Comment:
I suspect a new user would guess that `retry.doBackoff()` returns true if it ran, and false if it didn't. I think they'd be surprised to learn that it returns true if it should run a further time, and otherwise false.

I'd implement this as a `while(retry.doBackoff())`, where `retry.doBackoff()` returns true if it ran, and false if it didn't.

Reply:
I think the more surprising behavior would be that `retry.doBackoff()` does nothing the first time you call it, but after that it sleeps.

But I see your point, and I'm not sure what's "more correct"

Reply:
I do think `doBackoff()` should return true/execute the first time you call it. I'd only return false if it didn't run, which would be the last time it's called.

Reply:
The name doBackoff implies that it's doing something.
I don't see any sense in putting it at the start of the loop, just so it could do nothing the first time.

I agree that maybe the return value of "returns whether or not you can call it again" is unclear, but I don't think changing the semantics is the solution.

Maybe a comment? or something else?

Reply:
Nevermind my suggestion about `do`/`while` --> `while` (was based on the `function` approach we decided not to pursue).